### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by NikolaiSviridov**

### CODE REVIEW

File: src/main/kotlin/cli/gitMinersCLI/ComplexityCodeChangesCLI.kt [hunk 1]
```diff
@@ -78,16 +85,17 @@ class ComplexityCodeChangesCLI : GitMinerMultithreadedOneBranchCLI(
...
+        val dataProcessor = ComplexityCodeChangesDataProcessor(
             numOfMonthInPeriod = numOfMonth,
             numOfCommitsInPeriod = numOfCommits,
             changeType = changedTypeStringToEnum[changeType]!!,
             periodType = periodTypeStringToEnum[periodType]!!
         )
-        miner.run()
-        miner.saveToJson(resources)
+
+        val miner = ComplexityCodeChangesMiner(repository,branch,numThreads = numThreads)
```
Comment (vovak) [inline src/main/kotlin/cli/gitMinersCLI/ComplexityCodeChangesCLI.kt]: Please use "reformat code" before commit, or consider setting up a linter in CI.

File: src/main/kotlin/dataProcessor/DataProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package dataProcessor
+
+
+interface DataProcessor<D> {
```
Comment (vovak) [inline src/main/kotlin/dataProcessor/DataProcessor.kt:9]: This class is very generic, judging by the names. Please consider using more informative names and/or adding an explanation of its purpose.

File: src/main/kotlin/util/UtilFunctions.kt [hunk 3]
```diff
@@ -86,4 +100,19 @@ object UtilFunctions {
         return Di[Di.size - 1]
     }
 
+    fun runInThreadPoolWithExceptionHandle(threadPool: ExecutorService, tasks: List<Runnable>) {
```
Comment (vovak) [inline src/main/kotlin/util/UtilFunctions.kt:110]: Please consider Kotlin coroutines as an alternative to java.util.concurrent tooling. Coroutines are much easier to use after some getting used to

**Review by vovak — APPROVED** [review]
Thank you for this work! Feel free to merge when you find appropriate: there are only a few comments left after the discussion yesterday.

File: src/main/kotlin/cli/gitMinersCLI/ComplexityCodeChangesCLI.kt (hunk 1 above)
Comment (NikolaiSviridov) [reply src/main/kotlin/cli/gitMinersCLI/ComplexityCodeChangesCLI.kt]: Done

File: src/main/kotlin/dataProcessor/DataProcessor.kt (hunk 2 above)
Comment (NikolaiSviridov) [reply src/main/kotlin/dataProcessor/DataProcessor.kt:9]: Added comments.

File: src/main/kotlin/util/UtilFunctions.kt (hunk 3 above)
Comment (NikolaiSviridov) [reply src/main/kotlin/util/UtilFunctions.kt:110]: I'll try to use them in future versions.

**Pull request merged**